pkg/agent/flags/flag_value_duration.go

```diff
@@ -0,0 +1,103 @@
+package flags
+
+import (
+	"math"
+	"strconv"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/agent/flags/keys"
+)
+
+type durationOption func(*durationFlagValue)
+
+func WithOverride(override FlagValueOverride) durationOption {
+	return func(d *durationFlagValue) {
+		d.override = override
+	}
+}
+
+func WithDefault(defaultVal time.Duration) durationOption {
+	return func(d *durationFlagValue) {
+		d.defaultVal = int64(defaultVal)
+	}
+}
```

Comment:
You can probably roll these together into something that just appends to an array. Then iterate the array and take the first non-nil